### PR TITLE
Integrate auth feature to API

### PR DESCRIPTION
**Pull request by raendyelvicar**
Did you write good & comprehensive specs? ⭐️

What's this PR do ? ⭐️
Integrate FE login and register to API

What are the relevant tickets ? ⭐️
https://www.notion.so/prawito/Integrate-Auth-Page-User-with-API-2eb75942b4174407a082f967a853403e

Where should the reviewer start ?

How should this be manually tested ?

What is the Dockup URL ?

Any background context you want to provide ?

Any deployment note (migrations, new envs, rake tasks, cron) ?

Screenshots (if appropriate)
image

image

image

Definition of Done:

[ ] Specs / Tests are adequate ? ⭐️

[ ] Changes conform with code quality tools (Rubocop, Reek, Brakeman) ? ⭐️

[ ] Does this PR require Database changes ?

[ ] Does this PR require monitoring after deployment (Scout, logs, Sentry, etc) ?

[ ] Does this PR require a regression test by QA ? All fixes require a regression test.

[ ] Does this add new dependencies ? (network setup, new 3rd party services)

### CODE REVIEW

**Review by chaterinamf — APPROVED** [review]

**Review by herulhidayat — APPROVED** [review]

**Review by rahmatullah5 — CHANGES_REQUESTED** [review]

File: src/Actions/user.actions.js [hunk 1]
```diff
@@ -17,12 +20,14 @@ function login(email, password) {
             .then(
                 user => { 
                     dispatch(success(user));
-                    history.push('/dashboard');
+                    // history.push('/dashboard');
```
[the remark this answers is not in the document]
Comment (raendyelvicar) [reply src/Actions/user.actions.js:23]: Okaay kak

File: src/Actions/user.actions.js [hunk 2]
```diff
@@ -17,12 +20,14 @@ function login(email, password) {
...
                     window.location.replace('/dashboard');
+                    console.log(user);
```
[the remark this answers is not in the document]
Comment (raendyelvicar) [reply src/Actions/user.actions.js:25]: Okaay got it

File: src/Services/user.service.js [hunk 3]
```diff
@@ -30,17 +40,46 @@ function logout() {
...
-function register(user) {
+async function register(user) {
     const requestOptions = {
         method: 'POST',
         headers: { 'Content-Type': 'application/json' },
         body: JSON.stringify(user)
     };
 
-    return fetch("http://localhost:8000/api/register", requestOptions).then(handleResponse);
+    return await fetch("https://floating-journey-97236.herokuapp.com/api/v1/register", requestOptions).then(handleResponse);
```
[the remark this answers is not in the document]
Comment (raendyelvicar) [reply src/Services/user.service.js:50]: Okay kak, i'll try

**Comment by raendyelvicar** [comment]
All done

**Review by rahmatullah5 — APPROVED** [review]

**Pull request merged**